### ISSUE TITLE
Cannot tell if a partial fill is complete - documentation says differently

### ISSUE THREAD

**Issue by nzhammer** [coinbase/coinbase-pro-node#285]
When using [getOrder()](https://docs.gdax.com/#get-an-order) on a filled limit order and receiving back and Order that is a partial fill, I am not able to determine if that partial fill will be one of many or if the order is done and will only ever be partially filled.  I know that a limit order on the books can either:

1. Be filled completely by a single taker order
2. Be filled completely by multiple taker orders
3. Be filled partially by one or multiple taker orders

The first case is trivial to distinguish, whereas the second two are not.  Following the documentation on the GDAX API:

> Orders which are no longer resting on the order book, will be marked with the done status. There is a small window between an order being done and settled. An order is settled when all of the fills have settled and the remaining holds (if any) have been removed.

From this, I drew the conclusion that by using .getOrder(orderId) and checking result.status === 'done' && order.settled === true, I would know if my limit order was completely done filling (regardless of partial or complete fill). This turned out to be wrong. On multiple occasions I have called .getOrder(orderId), checked for a 'done' status and true settle, but have only received one partial fill of two, where calling .getOrder(orderId) a few seconds later resulted in the total fill.

I am guessing this is because my limit order filled with two partial fills and i called .getOrder(orderId) after the first part filled and before the second. The big problem here is that this leaves me in a position where I don't know if I received a partial fill and there are more fills on the way, or if that partial fill is the only fill that will happen and my order will never be completely filled.

**Is this a bug? Is there any way to know with 100% certainty that the filling of my order has completed, whether it was a single complete fill, several partial fills resulting in a complete fill, or partial fills resulting in an order that was not completely filled?**

Thanks!

**Comment by nzhammer** [comment]
Any ideas at least if this is a bug? Should a response with 'done' and 'settled' mean that all parts of the order have been filled?

**Comment by nzhammer** [comment]
Any updates on this? Just ran into another case of a partial fill where the order was settled, but we were only seeing a partial fill, contrary to the documentation.

**Comment by tkriplean** [comment]
I have the same problem of knowing whether more fills are coming or not after canceling an order. To ameliorate this issue, I use getOrder and getFills in conjunction with each other. Here's how: 

First, I poll getOrder until I get status == done or the order can't be found. Then I note the filled_size field returned from getOrder. If the order can't be found, I set filled_size to 0. 

Second, I request getFills for the order. I add up all the fills I received. If the total size of the fills from getFills matches getOrder (or is close enough), great, I treat the transaction as completed. If not, I repeat the two steps. 

I'm not convinced this works 100% of the time, and I wish Coinbase made it clearer. But I think it is superior to relying solely on getOrder. 

Would love to hear better suggestions!

**Comment by nzhammer** [comment]
@tkriplean Thanks for the response. I took a break from my project but now I'm back at it.

I see what you are saying, and I think I achieve the same thing by just noting how big the order was that I placed and checking getOrder()'s size to see if it was completely filled or not.

The problem Is mostly with knowing if another fill is coming when I realize that getOrder()'s size is not equal to my expected fill size.

For example, I see that the sizes are not equal. Should I cancel my order now and take the partial fill that I was able to get because it's never going to fully fill / won't fully fill until hours-days later? Or is it a transaction that is going to have completely filled in seconds - in which case I can wait? It is important for my program to move on quickly. If the order is filling near instantly in several fills, that's ok.. I can wait. If it's going to fill part now and that's it, I need to know.

I'm just looking for something that tells me 'a partial fill happened, but hold up a second because the rest is being filled now' or 'a partial fill happened and that's all you're going to get'.  Maybe this isn't reasonable because GDAX has no way of distinguising this, but based off of the documentation it seemed like using 'settled' / 'done' would make this acheivable.
